lib/ansible/modules/windows/win_optional_feature.ps1

```diff
@@ -0,0 +1,108 @@
+#!powershell
+
+# Copyright: (c) 2019, Carson Anderson <rcanderson23@gmail.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+#AnsibleRequires -CSharpUtil Ansible.Basic
+
+$spec = @{
+	options = @{
+		name = @{ type = "str"; required = $true }
+		state = @{ type = "str"; default = "present"; choices = @("absent", "present") }
+		source = @{ type = "str" }
+		include_parent = @{ type = "bool"; default = $false }
+	}
+	supports_check_mode = $true
+}
+
+$module = [Ansible.Basic.AnsibleModule]::Create($args, $spec)
+$module.Result.rc = 0
+
+$name = $module.Params.name
+$state = $module.Params.state
+$source = $module.Params.source
+$include_parent = $module.Params.include_parent
+
+$win_version = (Get-ItemProperty "HKLM:\SOFTWARE\Microsoft\Windows NT\CurrentVersion\").ProductName
+if ($win_version -notlike "Windows 10*") {
```

Comment:
You are better off doing a check for the cmdlet itself instead of the version, e.g.

```
if (-not (Get-Command -Name Enable-WindowsOptionalFeature -ErrorAction SilentlyContinue)) {
    $module.FailJson("blah")
}
```